pkg/vminetworkscontroller/vmi_controller.go

```diff
@@ -117,8 +117,8 @@ func (r *VirtualMachineInstanceReconciler) Reconcile(
 						return controllerruntime.Result{}, fmt.Errorf("let us be on the safe side and retry later")
 					}
 				}
-				if len(existingIPAMClaim.OwnerReferences) == 1 && existingIPAMClaim.OwnerReferences[0].UID == vm.UID {
-					r.Log.Info("found existing IPAMClaim belonging to this VM, nothing to do", "VM UID", vm.UID)
+				if len(existingIPAMClaim.OwnerReferences) == 1 && existingIPAMClaim.OwnerReferences[0].UID == ownerInfo.UID {
```

Comment:
We should add some e2e coverage here, this should have fail at CI.

Reply:
I agree. Please add a test for this. 

Still, good catch. This was a leftover of when I introduced VMI support.